api/python/cellxgene_census/src/cellxgene_census/_open.py

```diff
@@ -123,11 +135,20 @@ def open_soma(
     """
 
     if uri is not None:
-        return _open_soma({"uri": uri, "s3_region": None}, context)
+        return _open_soma({"uri": uri, "region": None, "provider": "file"}, context)
 
     if census_version is None:
         raise ValueError("Must specify either a census version or an explicit URI.")
 
+    mirrors = get_census_mirrors()
+    selected_mirror: CensusMirror
+    if mirror is not None:
+        if mirror not in mirrors:
+            raise KeyError("Mirror not found.")
```

Comment:
wrong error - should be a ValueError.  KeyError is explicitly restricted to a mapping type lookup failure.

Reply:
and yes, this points out that I used the wrong error in `get_census_version_description`.  Sigh!

Reply:
I will fix them both. Thanks for clarifying the extent of `KeyError`.